crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -233,6 +285,64 @@ impl IndexerStore for PgIndexerStore {
         ))
     }
 
+    async fn compose_full_transaction_response(
+        &self,
+        tx: Transaction,
+    ) -> Result<SuiTransactionFullResponse, IndexerError> {
+        let transaction: SuiTransaction =
+            serde_json::from_str(&tx.transaction_content).map_err(|err| {
+                IndexerError::InsertableParsingError(format!(
+                    "Failed converting transaction JSON {:?} to SuiTransaction with error: {:?}",
+                    tx.transaction_content, err
+                ))
+            })?;
+        let effects: SuiTransactionEffects = serde_json::from_str(&tx.transaction_effects_content).map_err(|err| {
+        IndexerError::InsertableParsingError(format!(
+            "Failed converting transaction effect JSON {:?} to SuiTransactionEffects with error: {:?}",
+            tx.transaction_effects_content, err
+        ))
+        })?;
+
+        let object_cache = ObjectProviderCache::new(Arc::new(self.clone()));
+        let balance_changes = get_balance_changes_from_effect(&object_cache, &effects).await?;
+
+        let sender = SuiAddress::from_str(tx.sender.as_str())?;
+        let object_changes = get_object_changes(
+            &object_cache,
+            sender,
+            &effects.modified_at_versions(),
+            effects.all_changed_objects(),
+            effects.all_deleted_objects(),
+        )
+        .await?;
+
+        let tx_digest: TransactionDigest = tx.transaction_digest.parse().map_err(|e| {
+            IndexerError::InsertableParsingError(format!(
+                "Failed to parse transaction digest {} : {:?}",
+                tx.transaction_digest, e
+            ))
+        })?;
+        let events = self.get_events(
+            EventFilter::Transaction(tx_digest),
+            None,
+            None,
+            /* descending_order */ false,
+        )?;
+
+        Ok(SuiTransactionFullResponse {
+            digest: tx_digest,
+            transaction,
+            raw_transaction: tx.raw_transaction,
+            effects,
+            confirmed_local_execution: tx.confirmed_local_execution,
+            timestamp_ms: tx.timestamp_ms as u64,
+            checkpoint: tx.checkpoint_sequence_number as u64,
+            events: SuiTransactionEvents { data: events.data },
+            object_changes: Some(object_changes),
```

Comment:
question: probably a premature optimization haha
But in the `get_transaction_with_options` endpoint, users can specify whether they want to see these changes. However in the PR I have at https://github.com/MystenLabs/sui/pull/9669 we'll always fetch the `SuiTransactionFullResponse` and then filter for what the caller wants. I think this is the case for other functions such as `get_object` in `pg_indexer_store`. Would we incur significant latency by always fetching `object_changes` and `balance_changes`? Would this matter to the end user?

Reply:
ah hold on i may have misunderstood this

Reply:
ah yeah I think in crates/sui-json-rpc/src/read_api.rs there is a `if opt.show_` that adds to `futures`